resources/js/components/whiteboards/Whiteboard.vue

```diff
@@ -851,6 +852,242 @@ const duplicateSelected = () => {
+        hiddenClipboardEl = document.createElement('textarea');
+        hiddenClipboardEl.id = 'hidden-clipboard';
+        hiddenClipboardEl.style.position = 'fixed';
+        hiddenClipboardEl.style.opacity = '0';
+        hiddenClipboardEl.style.pointerEvents = 'none';
+        hiddenClipboardEl.style.zIndex = '-1';
+        document.body.appendChild(hiddenClipboardEl);
+    }
+    return hiddenClipboardEl;
+};
+
+const copySelectedToClipboard = () => {
+    try {
+        if (!selectedIds.value.length) return;
+
+        const selectedShapes = shapes.value.filter(s => selectedIds.value.includes(s.id));
+        const json = JSON.stringify(selectedShapes);
+
+        const isSecure = window.isSecureContext && navigator.clipboard;
+
+        if (isSecure) {
+            navigator.clipboard.writeText(json).then(() => {
+                //console.log('Shapes copied to clipboard');
+            }).catch(err => {
+                //console.warn('Clipboard write failed:', err);
+                // fallback to hidden textarea
+                const el = ensureHiddenClipboard();
+                el.value = json;
+            });
+        } else {
+            // Localhost/dev fallback
+            const el = ensureHiddenClipboard();
+            //Testing link and text pasting on dev enviroment:
+            //el.value = 'This is text';
+            //el.value = json;
+            el.value = 'http://app.kanka.test:8081/w/1/entities/1650';
+            // Also try classic execCommand for good measure
+            try {
+                el.focus();
+                el.select();
+                document.execCommand('copy');
+            } catch (err) {
+                //console.warn('execCommand copy failed (fallback stored only):', err);
+            }
+
+            console.log('Shapes copied via hidden clipboard fallback');
+        }
+    } catch (err) {
+        console.error('Failed to copy shapes:', err);
+    }
+};
+const pasteFromClipboard = async () => {
+    try {
+        let text = '';
+
+        // Use modern API if possible
+        if (navigator.clipboard && window.isSecureContext) {
+            text = await navigator.clipboard.readText();
+        } else {
+            // Fallback for localhost/dev
+            const el = ensureHiddenClipboard();
+            text = el.value;
+        }
+
+        if (!text) return;
+
+        // Regex to detect /w/{{campaign_id}}/entities/{{id}}
+        const entityUrlRegex = /\/w\/\d+\/entities\/(\d+)/;
+        const match = text.match(entityUrlRegex);
+
+        if (match) {
+            const entityId = match[1];
+
+            try {
+                const res = await axios.get(`${props.entity}?entity_id=${entityId}`);
+                const entityData = res.data.data;
+                console.log(entityData);
+
+                // Create the entity shape
+                createEntityShape(entityData);
+            } catch (err) {
+                console.error('Failed to fetch entity:', err);
```

Comment:
Please tell the user (`window.showToast(message, error)`) that something went wrong rather than doing nothing visible to them.